tests/waveplate_stub_test.py

```diff
@@ -0,0 +1,94 @@
+from pnpq.devices.waveplate_stub import WaveplateStub
+from pnpq.errors import (
+    DeviceDisconnectedError,
+    WaveplateInvalidDegreeError,
+    WaveplateInvalidMotorChannelError,
+    WaveplateInvalidStepsError,
+)
+import pytest
+
+@pytest.mark.parametrize("f,argc", [("identify", 0), ("home", 0), ("auto_update_start", 0), ("auto_update_stop", 0), ("disable_channel", 1), ("enable_channel", 1), ("rotate", 1), ("getpos", 0), ("step_forward", 1), ("step_backward", 1), ("rotate_relative", 1), ("custom_home", 1), ("custom_rotate", 1)])
+def test_access_without_connection(f, argc):
+    wp = WaveplateStub()
+    with pytest.raises(DeviceDisconnectedError):
+        # Since the device is not connected, all methods should raise DeviceDisconnectedError
+        # For simplicity, 1 is passed as argument for all methods, the value should be small enough to not raise any other errors
+        # TODO: Refactor this to be more flexible in the future
+        arg = ",".join(["1"]*argc)
+        eval(f"wp.{f}({arg})")
+
+@pytest.fixture
+def c_wp():
```

Comment:
There's a reason I said `connected_waveplate` and not `c_wp`.

Short variable names are difficult to read and understand, especially when you are not familiar with the code. I am trying to slowly nudge our codebase towards using full English words.

For more on choosing variable names, see this summary of [_Clean Code_](https://dzone.com/articles/naming-conventions-from-uncle-bobs-clean-code-phil) by Robert Martin.